cmd/vfkit/main.go

```diff
@@ -113,10 +117,22 @@ func waitForVMState(vm *vz.VirtualMachine, state vz.VirtualMachineState) error {
 }
 
 func runVFKit(vmConfig *config.VirtualMachine, opts *cmdline.Options) error {
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
 	vzVMConfig, err := vf.ToVzVirtualMachineConfig(vmConfig)
 	if err != nil {
 		return err
 	}
+
+	if opts.UseGUI {
+		for _, gpuDev := range vmConfig.VirtioGPUDevices() {
+			if gpuDev.Device == config.VirtioGPUDisplayDevice {
+				gpuDev.UsesGUI = true
```

Comment:
I didn't make any changes to the VirtioGPU type. The `UsesGUI` attribute is still there for each Virtio-gpu device, and is automatically set to false. It can only be set to true if the user provides the `--gui` flag which is initially checked here.